catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -409,91 +489,126 @@ private void assertReplies(List<Mutable> replies) {
+                    emptyResponse);
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            LOGGER.debug("Error querying framework", e);
+        }
+        return framework;
+    }
+
+    private CatalogFramework createFramework(long count) {
 
         CatalogFramework framework = mock(CatalogFramework.class);
-        List<Result> results = new ArrayList<Result>();
-
-        for (int i = 0; i < COUNT; i++) {
-            Result result = mock(Result.class);
-
-            MetacardImpl metacard = new MetacardImpl();
-            metacard.setId("Metacard_" + i);
-            metacard.setTitle("Metacard " + i);
-            metacard.setLocation("POINT(" + i + " " + i + ")");
-            metacard.setType(BasicTypes.BASIC_METACARD);
-            metacard.setCreatedDate(TIMESTAMP);
-            metacard.setEffectiveDate(TIMESTAMP);
-            metacard.setExpirationDate(TIMESTAMP);
-            metacard.setModifiedDate(TIMESTAMP);
-            metacard.setContentTypeName("TEST");
-            metacard.setContentTypeVersion("1.0");
-            metacard.setTargetNamespace(URI.create(getClass().getPackage()
-                    .getName()));
-
-            when(result.getDistanceInMeters()).thenReturn(100.0 * i);
-            when(result.getRelevanceScore()).thenReturn(100.0 * (COUNT - i) / COUNT);
-            when(result.getMetacard()).thenReturn(metacard);
+        List<Result> results = new ArrayList<>();
 
+        for (int i = 0; i < count; i++) {
+            Result result = generateResult(i, count);
             results.add(result);
         }
 
         QueryResponse response = new QueryResponseImpl(mock(QueryRequest.class),
-                new ArrayList<Result>(),
-                COUNT);
+                new ArrayList<>(),
+                count);
         response.getResults()
                 .addAll(results);
 
+        QueryResponse emptyResponse = new QueryResponseImpl(mock(QueryRequest.class),
+                new ArrayList<>(),
+                0);
+
         try {
-            when(framework.query(any(QueryRequest.class))).thenReturn(response);
-        } catch (UnsupportedQueryException e) {
-            LOGGER.debug("Error querying framework", e);
-        } catch (SourceUnavailableException e) {
-            LOGGER.debug("Error querying framework", e);
-        } catch (FederationException e) {
+            when(framework.query(any(QueryRequest.class))).thenReturn(response, emptyResponse);
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
             LOGGER.debug("Error querying framework", e);
         }
         return framework;
     }
 
-    /**
-     * Helper methods for generating exception stacks
-     */
-    private Exception generateExceptionStack(int targetSize) {
-        return generateExceptionStack(targetSize, 0, null);
+    private Result generateResult(int value, long count) {
```

Comment:
✏️ Changing the name to "generateResultMock" would increase the readability of the code when this method is invoked. In addition, changing "count" to something more descriptive would help as well.